### PR TITLE
Feature: added state to useValidation

### PR DESCRIPTION
**Pull request by stackoverfloweth**

### CODE REVIEW

File: src/useValidation/useValidation.ts [hunk 1]
```diff
@@ -29,6 +31,10 @@ export function useValidation<T>(
   const valid = computed(() => error.value === '')
   const invalid = computed(() => !valid.value)
   const pending = ref(false)
+  const initialValue = unref(value)
```
Comment (pleek91) [inline src/useValidation/useValidation.ts:34]: What if it's an object or an array? 

Also validation doesn't happen unless mounted. But the value can change before then. So this won't match expectation in all cases

File: src/useValidation/useValidation.ts [hunk 2]
```diff
@@ -45,16 +51,27 @@ export function useValidation<T>(
     }
 
     pending.value = false
+    validated.value = true
```
Comment (pleek91) [inline src/useValidation/useValidation.ts:54]: I think this has value.

File: src/useValidation/useValidation.ts [hunk 3]
```diff
@@ -29,6 +31,10 @@ export function useValidation<T>(
   const valid = computed(() => error.value === '')
   const invalid = computed(() => !valid.value)
   const pending = ref(false)
+  const initialValue = unref(value)
+  const touched = ref(false)
+  const dirty = ref(false)
```
Comment (pleek91) [inline src/useValidation/useValidation.ts:36]: These can be easily determined by the user at implementation. And the composition doesn't use them internally. So I don't think these offer enough value.

**Comment by stackoverfloweth** [comment]
@pleek91 I only kept what you said had value and fixed some eslint warnings. I will make a follow-up change to prefect-design to replace it's `State` model with something more a kin to the type `UseValidation`

**Review by pleek91 — APPROVED** [review]

**Pull request merged**